src/v/redpanda/admin/aip_filter.cc

```diff
@@ -0,0 +1,371 @@
+/*
+ * Copyright 2025 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "redpanda/admin/aip_filter.h"
+
+#include "absl/strings/escaping.h"
+#include "absl/strings/numbers.h"
+#include "absl/time/time.h"
+#include "src/v/redpanda/admin/tests/aip_filter_test_messages.proto.h"
+
+#include <boost/spirit/home/x3.hpp>
+#include <fmt/format.h>
+
+#include <algorithm>
+#include <type_traits>
+
+namespace admin {
+
+namespace spirit_parser {
+namespace x3 = boost::spirit::x3;
```

Comment:
should we be concerned about this that is no longer maintained? https://github.com/boostorg/spirit/ I worry about relying on esoteric boost libraries. Looks like v4 is not yet ready for usage

Because I cannot really reason about the grammar stuff below. Like maybe we should consider https://github.com/foonathan/lexy?